inc/functions-shortcodes.php

```diff
@@ -215,3 +218,50 @@ function members_login_form_shortcode() {
 
 	return wp_login_form( array( 'echo' => false ) );
 }
+
+/**
+ * Filters the login redirect URL to send failed logins back to the
+ * referrer with a query arg of `login=failed`.
+ *
+ * @since 3.2.18
+ *
+ * @param string $redirect_to    The redirect destination URL.
+ * @param string $request        The request URL.
+ * @param object $user           The user object.
+ * @return string                The redirect URL.
+ */
+function members_login_redirect( $redirect_to, $request, $user ) {
+    // If there's a valid referrer, and it's not the default log-in screen
+    if ( ! isset( $_POST['members_redirect_to'] ) ) {
+        return $redirect_to;
+    } elseif ( empty( $user ) || is_wp_error( $user ) ) {
+        wp_redirect( str_replace('?login=failed', '', $_SERVER['HTTP_REFERER']) . '?login=failed' );
+        die;
+    } else {
+        return str_replace('?login=failed', '', $_SERVER['HTTP_REFERER']);
+    }
+}
+
+/**
+ * Filters the login form bottom output to add an error message if the login has failed.
+ *
+ * @since 3.2.18
+ *
+ * @return string The HTML to output below the login form.
+ */
+function members_login_form_bottom() {
+    if (
+        isset( $_SERVER['HTTP_REFERER'] ) &&
+        ! empty( $_SERVER['HTTP_REFERER'] ) &&
+        ! strstr( $_SERVER['HTTP_REFERER'],'wp-login' ) &&
+        ! strstr( $_SERVER['HTTP_REFERER'],'wp-admin' )
+    ) {
+        $output = '<input type="hidden" name="members_redirect_to" value="' . esc_attr( $_SERVER['HTTP_REFERER'] ) . '" />';
```

Comment:
@ThemeGravity I would just hard code this value to "1" or "true" or "yes". Rather than setting it to the referrer which may not always be available.

Reply:
@cartpauj I changed the value to 1